core/runtime/v2/shim/util_unix.go

```diff
@@ -177,3 +201,88 @@ func CanConnect(address string) bool {
 	conn.Close()
 	return true
 }
+
+func HybridVsockDialer(addr string, port uint64, timeout time.Duration) (net.Conn, error) {
+	timeoutCh := time.After(timeout)
+	// Do 10 retries before timeout
+	retryInterval := timeout / 10
+	for {
+		conn, err := net.DialTimeout("unix", addr, timeout)
+		if err != nil {
+			return nil, err
+		}
+		if _, err = conn.Write([]byte(fmt.Sprintf("CONNECT %d\n", port))); err != nil {
+			conn.Close()
+			return nil, err
+		}
+		errChan := make(chan error)
```

Comment:
suggest to use `make(chan error, 1)`, if not and it's timeout, the  following goroutine will be leaky.

Reply:
done